social_media_base/static/src/components/social_network_ads_account/social_network_ads_account.esm.js

```diff
@@ -0,0 +1,128 @@
+            loaderAds: false,
+        });
+        onWillStart(async () => {
+            await this._loadAdsAccount();
+        });
+    }
+
+    filter_ads(item, startDate, endDate, val_search) {
+        const created = DateTime.fromFormat(item.created, "dd/MM/yyyy");
+        const start_date = DateTime.fromFormat(startDate, "yyyy-MM-dd");
+        const end_date = DateTime.fromFormat(endDate, "yyyy-MM-dd");
+        return (
+            (start_date ? created >= start_date : false) &&
+            (end_date ? created <= end_date : false) &&
+            (val_search
+                ? (item.campaign.name
+                      ? item.campaign.name.includes(val_search)
+                      : false) ||
+                  (item.post.name ? item.post.name.includes(val_search) : false) ||
+                  item.status.includes(val_search)
+                : true)
+        );
+    }
+
+    onFilterAds({startDate, endDate, val_search}) {
+        if (val_search || startDate || endDate) {
+            this.social_state.socialAds = this.socialAdsAccount.ads.filter((item) => {
+                return this.filter_ads(item, startDate, endDate, val_search);
+            });
+        } else {
+            this.clearFilter();
+        }
+    }
+
+    /**
+     * Clears the filter and displays all ads again.
+     *
+     * When called, this method resets the `socialAds` state to the original
+     * list of ads retrieved from the server, effectively clearing any
+     * filtering criteria.
+     */
+    clearFilter() {
+        this.social_state.socialAds = this.socialAdsAccount.ads;
+    }
+
+    /**
+     * Gets the ads after applying the filter criteria.
+     *
+     * @returns {Object[]} - The ads after applying the filter criteria.
+     */
+    get ads() {
+        return this.social_state.socialAds;
+    }
+
+    /**
+     * Loads all ads again from the server.
+     *
+     * This method is triggered by the "Sync ads" button and is used to
+     * reload all ads from the server. It will clear any filtering criteria
+     * and display all ads again.
+     *
+     * @returns {Promise<void>}
+     */
+    async onUpdateAllAds() {
+        await this._loadAdsAccount();
+    }
+
+    /**
+     * Loads all ads from the server.
+     *
+     * This method is called when the user clicks on the "Sync ads" button.
+     * It will clear any filtering criteria, set the `loaderAds` state to
+     * `true`, and load all ads from the server. After loading the ads,
+     * it sets the `loaderAds` state to `false` and updates the component's
+     * state with the retrieved ads.
+     *
+     * @returns {Promise<void>}
+     */
+    async _loadAdsAccount() {
+        this.social_state.loaderAds = true;
+        const adsAccount = await this.ormService.call(
+            "social.network.account",
+            "load_ads_accounts",
+            [[]]
+        );
+        this.socialAdsAccount = adsAccount;
+        this.social_state.socialAds = adsAccount.ads;
```

Comment:
Good morning @edescalona . If you have the social_media_linkedin add-on uninstalled and you try to access the Configuration > Ads section, the following error occurs. 
![Captura desde 2025-04-30 14-39-31](https://github.com/user-attachments/assets/894092fe-29a5-470e-90a3-af942a9d785b)
I’ve had a quick look and noticed that the backend call without the social_media_linkedin add-on returns an empty dictionary. I assume JS treats that as an object, but since it’s empty the ads property doesn’t exist and evaluates to undefined.

```suggestion
        this.social_state.socialAds = adsAccount.ads || [] ;
```